lib/src/main/java/com/auth0/android/jwt/Claim.java

```diff
@@ -82,4 +82,14 @@
      * @throws DecodeException if the values inside the List can't be converted to a class T.
      */
     <T> List<T> asList(Class<T> tClazz) throws DecodeException;
+
+    /**
+     * Get this Claim as a Object of type T.
+     * If the value isn't of type Object, null will be returned.
```

Comment:
>  If the value isn't of type Object, null will be returned.

So if a user passes `Integer.class` they won't be able to obtain this claim as that? I'm thinking of generic scenarios where the type of the claim could be set dynamically and could end up being something not in the shape of a "JSON object". What are your thoughts?

Reply:
I removed the `value.isJsonObject()` check as I think it was unnecessary because it was better to throw `JsonSyntaxException`. And also as result it now works for Integer.class, Boolean.class, etc